.github/workflows/build_wheels.yml

```diff
@@ -42,24 +42,42 @@ env:
 
 jobs:
 
-  check_gcp_variables:
+  check_env_variables:
     timeout-minutes: 5
-    name: "Check GCP variables"
+    name: "Check environment variables"
     runs-on: ubuntu-latest
+    env:
+      EVENT_NAME: ${{ github.event_name }}
+      PY_VERSIONS_FULL: "cp37-* cp38-* cp39-* cp310-*"
     outputs:
-      gcp-variables-set: ${{ steps.check_gcp_variables.outputs.gcp-variables-set }}
+      gcp-variables-set: ${{ steps.check_env_variables.outputs.gcp-variables-set }}
+      py-versions-full: ${{ steps.set-py-versions.outputs.py-versions-full }}
+      py-versions-test: ${{ steps.set-py-versions.outputs.py-versions-test }}
     steps:
       - uses: actions/checkout@v3
       - name: "Check are GCP variables set"
         run: "./scripts/ci/ci_check_are_gcp_variables_set.sh"
-        id: check_gcp_variables
+        id: check_env_variables
         env:
           GCP_SA_EMAIL: ${{ secrets.GCP_SA_EMAIL }}
           GCP_SA_KEY: ${{ secrets.GCP_SA_KEY }}
           GCP_PYTHON_WHEELS_BUCKET: ${{ secrets.GCP_PYTHON_WHEELS_BUCKET }}
           GCP_PROJECT_ID: "not-needed-here"
           GCP_REGION: "not-needed-here"
           GCP_TESTING_BUCKET: "not-needed-here"
+      - name: Set Python Versions for different environments
+        id: set-py-versions
+        run: |
+          set -xeu
+          if [ $EVENT_NAME == "pull_request" ]; then
+            # run lowest and highest supported version on pull request
+            PY_VERSIONS_TEST="cp37-* cp310-*"
```

Comment:
maybe just run on highest version if this is not useful information on every PR?

Reply:
makes sense to me. Running on https://github.com/Abacn/beam/actions/runs/3595305093/